server/health.go

```diff
@@ -42,3 +64,18 @@ func (s *Server) checkCRIHealth(ctx context.Context, timeout time.Duration) erro
 
 	return nil
 }
+
+func (s *Server) cniPluginReadinessCheck(ctx context.Context) {
+	cniInitOnce.Do(func() {
+		go func() {
+			watcher := s.config.CNIPluginAddWatcher()
+			log.Infof(ctx, "Waiting for CNI plugin readiness.")
+			if ready := <-watcher; !ready {
+				log.Errorf(ctx, "CNI plugin not ready after waiting.")
+			} else {
+				log.Infof(ctx, "CNI plugin is ready.")
+				cniPluginInitialized = true
```

Comment:
I think this needs to either be an atomic.Bool or be protected by a lock

Reply:
I was thinking about it but `cniPluginInitialized` will not be access by multiple goroutines so I thought of keeping it as-is.

Reply:
it will be written by a separate one than it's read by right?

Reply:
ok, I see your point.